astropy_helpers/openmp_helpers.py

```diff
@@ -98,10 +144,44 @@ def add_openmp_flags_if_available(extension):
         os.chdir(start_dir)
 
     if using_openmp:
-        log.info("Compiling Cython extension with OpenMP support")
-        extension.extra_compile_args.append(compile_flag)
-        extension.extra_link_args.append(link_flag)
+        log.info("Compiling Cython/C/C++/ extension with OpenMP support")
+        if extension:
+            extension.extra_compile_args.extend(compile_flags)
+            extension.extra_link_args.extend(link_flags)
     else:
         log.warn("Cannot compile Cython extension with OpenMP, reverting to non-parallel code")
 
     return using_openmp
+
+_IS_OPENMP_ENABLED_SRC = """
+# Autogenerated by {packagetitle}'s setup.py on {timestamp!s}
+
+def is_openmp_enabled():
+    return {return_bool}
+"""[1:]
+
+def generate_openmp_enabled_py(packagename, srcdir='.'):
+    """
+    Utility for creating openmp_enabled.py::is_openmp_enabled()
+    used to determine, post build, whether the package was built
+    with or without OpenMP support.
+    """
+
+    if packagename.lower() == 'astropy':
+        packagetitle = 'Astropy'
+    else:
+        packagetitle = 'Astropy-affiliated package ' + packagename
```

Comment:
Not all packages using astropy-helpers are affiliated packages, so I would just set packagetitle = packagename